node/avn-service/src/ethereum_events_handler.rs

```diff
@@ -660,16 +699,22 @@ where
     };
     log::debug!("Eth-bridge instances found: {:?}", &instances);
     for (instance_id, instance) in instances {
-        // TODO check if there is a web3 connection for the instance chain_id. If not, skip the
-        // instance
-
         let result = &config
             .client
             .runtime_api()
             .query_active_block_range(config.client.info().best_hash, instance_id)
             .map_err(|err| format!("Failed to query bridge contract: {:?}", err))?;
 
-        let web3_data_mutex = config.web3_data_mutex.lock().await;
+        let web3_data_lock = match initialize_web3_connection_for_instance(config, &instance).await
+        {
+            Ok(web3_data) => web3_data,
+            Err(e) => {
+                log::error!("Failed to initialize web3 connection for instance: {:?}", e);
+                continue
```

Comment:
do we want to continue here?

Reply:
yes, we have other instances that are waiting.